pages/[id]/edit.js

```diff
@@ -0,0 +1,177 @@
+  };
+
+  const handleSubmit = (e) => {
+    setOpened(!opened);
+    setFishName("");
+    setFishWeight("");
+    setFishLength("");
+    setFishLocation("");
+    sendToServer();
+  };
+
+  return (
+    <div>
+      <Map setLatLng={setLatLng} latlng={latlng} fetchedData={fetchedData} />
+      <StyledForm onSubmit={handleSubmit}>
+        {opened ? (
+          <Modal open={opened} close={() => setOpened(!opened)}>
+            <p>You&apos;re input has been added to the List!</p>
+            <Link aria-label="Browse the list page" href="/List" passHref>
+              <Anchor active={pathname === "../List"}>
+                <p>Move to the List!</p>
+              </Anchor>
+            </Link>
+          </Modal>
+        ) : null}
+
+        <StyledField>
+          <StyledLabel htmlFor="name">Fish Name: </StyledLabel>
+          <input
+            type="text"
+            id="name"
+            name="name"
+            minLength="3"
+            maxLength="15"
+            placeholder="z.B. Lachs"
+            onChange={(e) => setFishName(e.target.value)}
+            pattern="^(?!^ +$)([\w -&]+)$"
+            required
+          />
+          <StyledLabel htmlFor="weight">Weight in kg: </StyledLabel>
+          <input
+            type="number"
+            id="weight"
+            name="weight"
+            step="0.10"
+            min=".50"
+            max="25"
+            placeholder="z.B. 0.70"
+            oonChange={(e) => setFishWeight(e.target.value)}
+            required
+          />
+          <StyledLabel htmlFor="length">Length in cm: </StyledLabel>
+          <input
+            type="number"
+            id="length"
+            name="length"
+            placeholder="z.B. 10"
+            step="10"
+            min="10"
+            max="200"
+            onChange={(e) => setFishLength(e.target.value)}
+            required
+          />
+          <StyledLabel htmlFor="location">Location: </StyledLabel>
+          <input
+            type="text"
+            id="location"
+            name="location"
+            minLength="3"
+            maxLength="15"
+            placeholder="z.B. Kristiansand"
+            onChange={(e) => setFishLocation(e.target.value)}
+            pattern="^(?!^ +$)([\w -&]+)$"
+            required
+          />
+          <DatePickerContainer>
+            <DatePicker
+              selected={startDate}
+              onChange={(date) => setStartDate(date)}
+              showTimeSelect
+              timeFormat="HH:mm"
+              timeIntervals={15}
+              timeCaption="time"
+              dateFormat="h:mm aa d MMMM, yyyy "
+              withPortal
+            />
+          </DatePickerContainer>
+        </StyledField>
+
+        <StyledButton type="submit">Update</StyledButton>
+      </StyledForm>
+    </div>
+  );
+}
+
+export async function getServerSideProps(ctx) {
+  const res = await fetch(`${apiUrl(ctx)}/formdata/${ctx.query.id}`);
+  const data = await res.json();
```

Comment:
Do not fetch from your API inside getServerSideProps. Read the data here directly from your service / database instead!